cli/internal/run/real_run.go

```diff
@@ -150,6 +150,22 @@ func RealRun(
 		base.UI.Error(err.Error())
 	}
 
+	// When continue on error is enabled don't register failed tasks as errors
+	// and instead must inspect the task summaries.
+	if ec.rs.Opts.runOpts.continueOnError {
+		for _, summary := range runSummary.RunSummary.Tasks {
+			if childExit := summary.Execution.ExitCode(); childExit != nil {
+				childExit := *childExit
+				if childExit < 0 {
+					childExit = -childExit
+				}
+				if childExit > exitCode {
+					exitCode = childExit
+				}
+			}
+		}
+	}
```

Comment:
Hmm, when we use `--continue` does `errs := engine.Execute(visitorFn, execOpts)` not return the ChildExitErr? I feel like it would be a better fix to always return the ChildExit from the task and then choose what to do about it based on the `--continue` flag. (we don't have to block on this if it's urgent to get out)

Reply:
> Hmm, when we use --continue does errs := engine.Execute(visitorFn, execOpts) not return the ChildExitErr?

Correct, we swallow the error and just log a warning along with populating the task execution summary.

> I feel like it would be a better fix to always return the ChildExit from the task and then choose what to do about it based on the --continue flag

So doing this involves exposing the run options to the engine which is a little more plumbing that I wanted to do since this bug is giving front issues.